src/game/Laptop/IMP_SkillTraits.cc

```diff
@@ -587,64 +605,9 @@ void AddSelectedSkillsToSkillsList()
 		if( gfSkillTraitQuestions[ uiCnt ] )
 		{
 			//switch on the trait, and add it to the list
-			switch( uiCnt )
-			{
-				case IMP_SKILL_TRAITS__LOCKPICK:
-					AddSkillToSkillList( LOCKPICKING );
-					break;
-
-				case IMP_SKILL_TRAITS__HAND_2_HAND:
-					AddSkillToSkillList( HANDTOHAND );
-					break;
-
-				case IMP_SKILL_TRAITS__ELECTRONICS:
-					AddSkillToSkillList( ELECTRONICS );
-					break;
-				
-				case IMP_SKILL_TRAITS__NIGHT_OPS:
-					AddSkillToSkillList( NIGHTOPS );
-					break;
-				
-				case IMP_SKILL_TRAITS__THROWING:
-					AddSkillToSkillList( THROWING );
-					break;
-				
-				case IMP_SKILL_TRAITS__TEACHING:
-					AddSkillToSkillList( TEACHING );
-					break;
-				
-				case IMP_SKILL_TRAITS__HEAVY_WEAPONS:
-					AddSkillToSkillList( HEAVY_WEAPS );
-					break;
-				
-				case IMP_SKILL_TRAITS__AUTO_WEAPONS:
-					AddSkillToSkillList( AUTO_WEAPS );
-					break;
-				
-				case IMP_SKILL_TRAITS__STEALTHY:
-					AddSkillToSkillList( STEALTHY );
-					break;
-				
-				case IMP_SKILL_TRAITS__AMBIDEXTROUS:
-					AddSkillToSkillList( AMBIDEXT );
-					break;
-				
-				case IMP_SKILL_TRAITS__KNIFING:
-					AddSkillToSkillList( KNIFING );
-					break;
-				
-				case IMP_SKILL_TRAITS__ROOFTOP_SNIPING:
-					AddSkillToSkillList( ONROOF );
-					break;
-				
-				case IMP_SKILL_TRAITS__CAMO:
-					AddSkillToSkillList( CAMOUFLAGED );
-					break;
-				
-				case IMP_SKILL_TRAITS__MARTIAL_ARTS:
-					AddSkillToSkillList( MARTIALARTS );
-					break;
-			}
+			auto impTrait = static_cast<ImpSkillTrait>(uiCnt);
+			auto skillTrait = skillTraitsMapping.at(impTrait);
```

Comment:
They can also pick none, which should throw an exception here. Right?

Reply:
No, this loop does not reach `IMP_SKILL_TRAITS__NONE`:

https://github.com/ja2-stracciatella/ja2-stracciatella/blob/52c756b4988555d694b26d41e3685f385ab715fc/src/game/Laptop/IMP_SkillTraits.cc#L602

 `IMP_SKILL_TRAITS__NONE` is handled here:

https://github.com/ja2-stracciatella/ja2-stracciatella/blob/52c756b4988555d694b26d41e3685f385ab715fc/src/game/Laptop/IMP_SkillTraits.cc#L358-L359